File: basemodule/base/src/main/java/com/zsy/basemodule/base/utils/GSON.java
package com.zsy.basemodule.base.utils;

import com.google.gson.Gson;

import java.lang.ref.WeakReference;

public class GSON {

    private static WeakReference<Gson> instance;

    private GSON() {

    }

    public static Gson get() {
        if (instance == null || instance.get() == null) {
            Gson gson = new Gson();
            instance = new WeakReference<>(gson);
        }
        return instance.get();
    }

}

File: basemodule/base/src/main/java/com/zsy/basemodule/base/utils/GSON.kt
package com.zsy.basemodule.base.utils

import com.google.gson.Gson
import java.lang.ref.WeakReference

object GSON {
    private var instance: WeakReference<Gson?>? = null
    fun get(): Gson? {
        if (instance == null || instance!!.get() == null) {
            val gson = Gson()
            instance = WeakReference(gson)
        }
        return instance!!.get()
    }
}